rs/boundary_node/rate_limits/canister/certification.rs

```diff
@@ -0,0 +1,174 @@
+use std::{cell::RefCell, rc::Rc};
+
+use ic_asset_certification::{Asset, AssetConfig, AssetEncoding, AssetFallbackConfig, AssetRouter};
+use ic_cdk::api::{data_certificate, set_certified_data};
+use ic_http_certification::{HeaderField, HttpCertificationTree, HttpRequest, HttpResponse};
+
+use crate::{
+    metrics::{export_metrics_as_http_response, with_metrics_registry},
+    state::with_canister_state,
+};
+
+thread_local! {
+    static HTTP_TREE: Rc<RefCell<HttpCertificationTree>> = Default::default();
+
+    // initializing the asset router with an HTTP certification tree is optional.
+    // if direct access to the HTTP certification tree is not needed for certifying
+    // requests and responses outside of the asset router, then this step can be skipped.
+    static ASSET_ROUTER: RefCell<AssetRouter<'static>> = RefCell::new(AssetRouter::with_tree(HTTP_TREE.with(|tree| tree.clone())));
+}
+
+// Handlers
+pub fn serve_asset(request: &HttpRequest<'static>) -> HttpResponse<'static> {
+    let path = match request.get_path() {
+        Ok(path) => path,
+        Err(err) => {
+            return HttpResponse::builder()
+                .with_status_code(400)
+                .with_body(
+                    format!("Malformed request: {:?}", err.to_string())
+                        .as_bytes()
+                        .to_vec(),
+                )
+                .build()
+        }
+    };
+
+    match path.as_str() {
+        // Serve non-certified metrics
+        "/metrics" => with_canister_state(|state| {
+            with_metrics_registry(|registry| export_metrics_as_http_response(registry, state))
+        }),
+        // Serve all other certified assets
+        _ => ASSET_ROUTER.with_borrow(|asset_router| match data_certificate() {
+            Some(certificate) => match asset_router.serve_asset(certificate.as_slice(), request) {
+                Ok(response) => response,
+                Err(err) => HttpResponse::builder()
+                    .with_status_code(500)
+                    .with_body(
+                        format!(
+                            "Failed to serve asset for path {}: {:?}",
+                            path.as_str(),
+                            err
+                        )
+                        .as_bytes()
+                        .to_vec(),
+                    )
+                    .build(),
+            },
+            None => HttpResponse::builder()
+                .with_status_code(500)
+                .with_body(b"No data certificate available".to_vec())
+                .build(),
+        }),
+    }
+}
+
+pub fn certify_assets(assets: Vec<Asset<'static, 'static>>) {
+    // Define the asset certification configurations.
+    let encodings = vec![
```

Comment:
There are no Brotli and Gzip encodings being added for any files, so either this config should be removed, or the encoded versions of those files would need to be added. To add the encoded versions of those files you'd need to include Brotli and Gzip encoding libraries in the canister and then encode the assets and add 3 versions of each asset, the original, the brotli encoded and the gzipped version. I only think it's worth the effort for static assets like JS, CSS, HTML etc, so I'd recommend just removing it.

Reply:
i'll remove it